src/xgov-dapp/src/hooks/use-applications-query.ts

```diff
@@ -0,0 +1,15 @@
+import * as algokit from '@algorandfoundation/algokit-utils'
+import { useQuery } from '@tanstack/react-query'
+
+const indexer = algokit.getAlgoIndexerClient({
+  server: import.meta.env.VITE_INDEXER_SERVER,
+  port: import.meta.env.VITE_INDEXER_PORT,
+  token: import.meta.env.VITE_INDEXER_TOKEN,
+})
+export default function useApplicationsQuery(addresses: string[]) {
```

Comment:
where is this used?

Reply:
It's an HTTP endpoint which should be used with the global state hooks but requires refactoring of the VotingRoundContract api. The `applicationsByCreator` need to be passed as props

https://github.com/algorandfoundation/nft_voting_tool/blob/954188991ff04c77dfbaa55370cddf7086124962/src/dapp/src/shared/VotingRoundContract.ts#L70

We can leave it there for now and use it down the road when we are implementing the hooks